### PR TITLE
"Disable Master Key Pair" tutorial

### PR DESCRIPTION
**Pull request by mDuo13**
This tutorial walks through the process of disabling an account's master key pair. This complements the existing tutorials to [Assign a Regular Key Pair](https://xrpl.org/assign-a-regular-key-pair.html) and [Set Up Multi-Signing](https://xrpl.org/set-up-multi-signing.html).

For the first pass, I've provided only examples using the rippled APIs directly. As a future feature, it would be great to have examples using ripple-lib and potentially other client-side libraries.

For those with Ripple VPN access, you can preview the changes here: https://developers-staging.ripple.com/disable-master-key-pair.html

### CODE REVIEW

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+
+**To disable the master key pair, you must use the master key pair.** However, you can _re-enable_ the master key pair using any other method of authorizing transactions.
+
+## Prerequisites
+
+To disable the master key pair for an account, you must meet the following prerequisites:
+
+- You must have an XRP Ledger [account](accounts.html) and you must be able to sign and submit transactions from that account using the master key pair. See also: [Set Up Secure Signing](set-up-secure-signing.html).
+    - Generally, you need to know account's master seed value to sign this transaction. A seed value is commonly represented as a [base58][] value starting with "s", such as `sn3nxiW7v8KXzPzAqzyHXbSSKNuN9`. A notable exception is that a [dedicated signing device](set-up-secure-signing.html#use-a-dedicated-signing-device) stores the seed value securely so you don't need to know it.
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md]: you need to know account's - add "a"

File: content/tutorials/manage-account-settings/assign-a-regular-key-pair.md [hunk 2]
```diff
@@ -394,6 +394,8 @@ Note that the response contains a `hash` of the transaction, which you can use t
 
 ## 3. Verify the Regular Key Pair
 
+At this point, the regular key pair is assigned to your account and you should be able to send transactions using the regular key pair. However, it is important to verify that you did not make a mistake before you take any additional steps such as [disabling the master key pair](disable-master-key-pair.html).
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/assign-a-regular-key-pair.md]: However, to prevent losses, it is important that you verify ...
(I wanted this warning to jump out a bit more, and also make clear the reader has to be the one to verify.)

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 3]
```diff
@@ -0,0 +1,497 @@
+# Disable Master Key Pair
+
+This page describes how to disable the [master key pair](cryptographic-keys.html) that is mathematically associated with an [account](accounts.html)'s address. You may wish to do this if you believe your account's master key pair has been compromised, or if you want to make [multi-signing](multi-signing.html) the _only_ way to submit transactions from your account.
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md]: "You may wish ..." -
arbitrary, but I suggest:
You may want to do this if you think that your account's master key pair may have been compromised ...
(I went for stronger language here, but perhaps it should be stronger. Leave off the original qualifier, and say
You should do this if you think that your account's master key pair may have been compromised ...

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+
+**Warning:** Disabling the master key pair removes one method of [authorizing transactions](transaction-basics.html#authorizing-transactions). You should be sure you can use one of the other ways of authorizing transactions, such as with a regular key or by multi-signing, before you disable the master key pair. (For example, if you [assigned a regular key pair](assign-a-regular-key-pair.html), make sure that you can successfully submit transactions with that regular key.) Due to the decentralized nature of the XRP Ledger, no one can restore access to your account if you cannot use the remaining ways of authorizing transactions.
+
+**To disable the master key pair, you must use the master key pair.** However, you can _re-enable_ the master key pair using any other method of authorizing transactions.
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md:7]: Under what circumstances would you re-enable the master key pair, if you are concerned that it has been compromised? Do you get a different master key pair when you re-enable it? (I assume that must be the case, as re-enabling the same pair doesn't sound sensible.)

File: content/tutorials/manage-account-settings/disable-master-key-pair.md (hunk 1 above)
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md]: Arbitrary again, but I would suggest breaking out these two scenarios (knowing the seed, and having a dedicated device).
* You have an XRP Ledger account ...
* You have access to the account's master seed value to sign this transaction. A seed value is commonly represented as a [base58][] value starting with "s", such as `sn3nxiW7v8KXzPzAqzyHXbSSKNuN9`. You either:
    * Know the account's master seed value, or
   * You have a [dedicated signing device](set-up-secure-signing.html#use-a-dedicated-signing-device) that stores the seed value securely so you don't need to know it.

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+## Prerequisites
+
+To disable the master key pair for an account, you must meet the following prerequisites:
+
+- You must have an XRP Ledger [account](accounts.html) and you must be able to sign and submit transactions from that account using the master key pair. See also: [Set Up Secure Signing](set-up-secure-signing.html).
+    - Generally, you need to know account's master seed value to sign this transaction. A seed value is commonly represented as a [base58][] value starting with "s", such as `sn3nxiW7v8KXzPzAqzyHXbSSKNuN9`. A notable exception is that a [dedicated signing device](set-up-secure-signing.html#use-a-dedicated-signing-device) stores the seed value securely so you don't need to know it.
+- Your account must have at least one method of authorizing transactions other than the master key pair. In other words, you must do one or both of the following:
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md:16]: you must do one or both of the following: - technically it is possible to disable the master-key pair and lose your value, if you haven't done this, right? I know you discuss this elsewhere, but because it's so significant, I'd rephrase this to indicate that if you don't do one of the following two things, you will lose your value.

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+    - [Assign a Regular Key Pair](assign-a-regular-key-pair.html).
+    - [Set Up Multi-Signing](set-up-multi-signing.html).
+
+## Steps
+
+{% set n = cycler(* range(1,99)) %}
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md:22]: Nice! I didn't know about this technique.

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+
+### {{n.next()}}. Construct Transaction JSON
+
+Prepare an [AccountSet transaction][] from your account with the field `"SetValue": 4`. This is the value for the account-set-flag Disable Master (asfDisableMaster). The only other required fields for this transaction are the required [common fields](transaction-common-fields.html). For example, if you leave off the [auto-fillable fields](transaction-common-fields.html#auto-fillable-fields), the following transaction instructions are enough:
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md]: account-set-flag - do you style this?

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+
+```json
+{
+  "TransactionType": "AccountSet",
+  "Account": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
+  "SetFlag": 4
+}
+```
+
+It is strongly recommended to also provide the `LastLedgerSequence` field so that you can [reliably get the outcome of the transaction in a predictable amount of time](reliable-transaction-submission.html).
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md]: This recommendation could be in the form of a tip or have a recommendation icon. Otherwise, I'd suggest to make the recommendation more forceful.
Recommendation: Include the the `LastLedgerSequence` field so that you can [reliably get the outcome of the transaction in a predictable amount of time](reliable-transaction-submission.html).

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 9]
```diff
@@ -0,0 +1,497 @@
...
+   }
+}
+```
+
+<!-- MULTICODE_BLOCK_END -->
+
+Look for `"status": "success"` to indicate that the server successfully signed the transaction. If you get `"status": "error"` instead, check the `error` and `error_message` fields for more information. Some common possibilities include:
+
+- `"error": "badSecret"` usually means you made a typo in the `secret` of the request.
+- `"error": "masterDisabled"` means this address's master key pair is _already_ disabled.
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md:172]: If it has been already disabled, is there a way to get a log of when this happened?

File: content/tutorials/manage-account-settings/assign-a-regular-key-pair.md [hunk 10]
```diff
@@ -394,6 +394,8 @@ Note that the response contains a `hash` of the transaction, which you can use t
...
+
 To verify that your account has the regular key pair set correctly, submit an [AccountSet transaction][] from your account, signing it with the regular private key you assigned to your account in step 2.
 
 As discussed in step 2, transmitting your master private key is dangerous. It is equally risky to transmit your regular private key. Therefore, we'll complete this transaction in two steps to keep transaction signing separate from transaction submission to the network.
```
Comment (LoisRP) [inline content/tutorials/manage-account-settings/assign-a-regular-key-pair.md]: "omplete this transaction in two steps " - I think the two separate steps are Sign your transaction and Submit your transaction, but it took a few extra hops to figure that out. So I might suggest:
Therefore, we'll complete this transaction with two separate steps--signing your transaction and submitting your transaction--to keep transaction signing separate from transaction submission to the network.

**Review by LoisRP — COMMENTED** [review]
(I still have to give this a try, but I am submitting these comments for now.)

File: content/tutorials/manage-account-settings/disable-master-key-pair.md (hunk 4 above)
Comment (mDuo13) [reply content/tutorials/manage-account-settings/disable-master-key-pair.md:7]: You cannot get a new master key pair; it's mathematically linked to your address.

You would not re-enable the master key pair if it has been compromised. However, if that is not the reason you disabled it, there are situations where you'd want to re-enable it. For example, if you don't want to use multi-signing anymore, or if you disabled your master key by accident. Or maybe you disabled the master key as a precaution but after a careful investigation you concluded that it was not compromised.

There are some small advantages of having the master key enabled even if you aren't actively using it. The biggest is probably that you can use it as a "backup" key to regain control of your account even if your regular key gets compromised. (The master key can change the regular key but a regular key can't disable the master key. If a malicious user learns only your regular private key, they can't lock you out as long as you can still use the master key.)

File: content/tutorials/manage-account-settings/disable-master-key-pair.md (hunk 1 above)
Comment (mDuo13) [reply content/tutorials/manage-account-settings/disable-master-key-pair.md]: Funny, I thought I had done this initially but apparently I changed it in while copy-editing. Anyway, I've now split it up slightly differently, but basically the same idea.

File: content/tutorials/manage-account-settings/disable-master-key-pair.md (hunk 5 above)
Comment (mDuo13) [reply content/tutorials/manage-account-settings/disable-master-key-pair.md:16]: No, it is not possible. The XRP Ledger does not allow a transaction to remove the last method of authorizing transactions from an account.

If you have neither assigned a regular key pair or set up multi-signing, the AccountSet to disable the master key pair fails with `tecNO_ALTERNATIVE_KEY`. I've added an extra reminder about this to the stage in the tutorial where that error would come up.

File: content/tutorials/manage-account-settings/disable-master-key-pair.md (hunk 7 above)
Comment (mDuo13) [reply content/tutorials/manage-account-settings/disable-master-key-pair.md]: No, it doesn't have specific code meaning. (It's just the explanation for what "asf" means).

I've changed the punctuation to hopefully make that more clear.

File: content/tutorials/manage-account-settings/disable-master-key-pair.md (hunk 9 above)
Comment (mDuo13) [reply content/tutorials/manage-account-settings/disable-master-key-pair.md:172]: There may not be. You have to look at the account's transaction history. On the Testnet, there are no (known) full history servers so if it didn't happen recently enough to be in the available history, you can't really know when that happened.

File: content/tutorials/manage-account-settings/assign-a-regular-key-pair.md (hunk 10 above)
Comment (mDuo13) [reply content/tutorials/manage-account-settings/assign-a-regular-key-pair.md]: I've rephrased both instances of this to link to the tutorial for setting up secure signing instead. (That tutorial didn't exist when the "Assign a Regular Key Pair" tutorial was originally written.)

File: content/tutorials/manage-account-settings/disable-master-key-pair.md [hunk 11]
```diff
@@ -0,0 +1,500 @@
...
+        }
+    }
+}
+
+```
+
+*Commandline*
+
+```sh
+Loading: "/home/mduo13/.config/ripple/rippled.cfg"
```
Comment (intelliot) [inline content/tutorials/manage-account-settings/disable-master-key-pair.md]: OK to leave the references to `mduo13` here?

**Review by intelliot — APPROVED** [review]
Looks good to me. In the future we could expand the tutorial to use a supported "offline" signing method.

File: content/tutorials/manage-account-settings/disable-master-key-pair.md (hunk 11 above)
Comment (mDuo13) [reply content/tutorials/manage-account-settings/disable-master-key-pair.md]: Should change it to match the standard example path.
